processor/src/test/java/org/mapstruct/ap/test/targetthis/ConflictingButIgnoredMapper.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.targetthis;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface ConflictingButIgnoredMapper {
+
+    ConflictingButIgnoredMapper INSTANCE = Mappers.getMapper( ConflictingButIgnoredMapper.class );
+
+    @Mapping( target = ".", source = "customer.item" )
+    @Mapping( target = "id", source = "customer.item.id", ignore = true )
```

Comment:
I'm really doubting whether we should do this.. have a way to ignore properties that occur in a next " target this " mapping.. To avoid conflicts the user should in such cases revert to fully mapping his intent. Lets consider target this as a convenience. Perhaps we should even allow it to be used only once in a set of mappings (at the other hand that might be too restricting).

Reply:
I am not really following. What shouldn't we allow? I don't see the need for the `source` here, since the ignore would apply on the target, so the source is obsolete. Or am I understanding something wrong?

Reply:
The two sources are identical and contain both the same properties: id and status. You need to tell which one to ignore.

Reply:
Or, alternatively you don't allow to create this situation by only allowing one "." Mapping

Reply:
Or you do allow it, but if there's a conflict the user needs to resolve this with by fully specifying mappings the normal way

Reply:
We could allow the multiple `.`. That could be the same as multi source parameters. However, you should not define the `ignore`. You should define how you want to map the `id` and status. 

So if you want the status from the customer and the id from the customer then you do.

```
@Mapping( target = "status", source = "customer.item.status" )
@Mapping( target = "id", source = "item.id" )
```

If that is too complex then let's just allow one `.`

Reply:
so.. that does not complete resolve the issue.. What if the user specifies:
```java
@Mapping( target = "status", source = "customer.item.status" )
@Mapping( target = ".", source = "item" )
```
?
Should I report an error (since the "." introduces a second mapping to status)?

Reply:
Honestly, for me this point "." seems little strange. I don't know why, but... I'm not sure about my understanding this discussion, but I think to report error is better.

Reply:
@Captain1653 .. what it in essence means is:
`Mapping( target = ".", source = "item.*" )`
so all the properties of item are mapped to the properties within target.

Reply:
Honestly, for me this poit "." seems  little strange. I don't know why, but it is. I'm not sure about good understanding this situation, but I think to report an error is better.

Reply:
@sjaakd  It's only my opinion, but it's not obvious for me.  I mean... I prefer explicit mapping. This approach is getting like SPEL from Spring. Too much magic  in the string, but maybe I'm wrong. Sure, it's only my opinion and I'll aprove it if you think we need it.

Reply:
Next question about this  point ".". Should it work recursively? Which depth?

Reply:
> t's only my opinion, but it's not obvious for me. I mean... I prefer explicit mapping. This approach is getting like SPEL from Spring. Too much magic in the string, but maybe I'm wrong. Sure, it's only my opinion and I'll aprove it if you think we need it.

It is a feature request and it does make kind of sense (it safes a lot of explicit mappings and is in essence name based mapping on nested level). So for instance: it is already possible to do this (if the target were to be nested as well)

```java
@Mapping( target = "item", source = "customer.item" ).
```

But I share your opinion that should limit its use. I guess it makes sense to only have one "." mapping since you cannot have multiple of the above like this..

```java
@Mapping( target = "item", source = "customer.item" ).
@Mapping( target = "item", source = "order.item" ).

```

Reply:
Yes, you are right. Only one point. If we have more than one point then to throw error. It's seems good.

Reply:
I think only allowing one `.` is too limiting. The use case of mapping a class with multiple nested classes to a fully flattened structure is not uncommon.

Reply:
but it creates inconsistencies and stuff that is hard to explain in the api. So as always.. its a trade-off.

Reply:
>so.. that does not complete resolve the issue.. What if the user specifies:
>
>```
>@Mapping( target = "status", source = "customer.item.status" )
>@Mapping( target = ".", source = "item" )
>```
>?
>Should I report an error (since the "." introduces a second mapping to status)?

If the user specifies this then status is already mapped and would be ignored when "." is used. I would say that "." should mean the rest of the unmapped properties. Since you might want to map certain properties in a different way.

All this means that there is no second mapping to status since "." means the rest (at least that's how I look at it conceptually).

Regarding multiple ".". Currently I see "." as a shortcut to saying mapping to the target. It doesn't have to be "." it can also be "<TARGET>", this is just an implementation detail.

In the end you could achieve the same with the current api by defining multiple source parameters.

Reply:
The multiple source parameters is a compelling argument.

 Lets do it like this, first we merge this PR. 

Next we build upon the implementation and allow multiple ".". . It could be just as simple as changing the logic in mapper reference from one targetThis to a list of targetThis (and have some smart logic at the call site, but that's similar than the name based below.